.github/workflows/release_workflow.yml

```diff
@@ -0,0 +1,104 @@
+name: CAS Release Workflow
+
+# Controls when the action will run.
+# Triggers the workflow on push or pull request events
+on: [push, workflow_dispatch]
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # This workflow contains a single job called "build"
+  tox-python-build:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python: [ '3.7', '3.8', '3.9', '3.10' ]
+
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v2
+
+      - name: Setup Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python }}
+
+      - name: Install Dev Requirments
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r dev-requirements.txt
+
+      - name: Install Tox
+        run: pip install tox
+
+      - name: Run Tox lint commands
+        # Run tox using the version of Python in `PATH`
+        run: tox -e lint
+
+      - name: Run Pytest through Tox
+        # Run tox using the version of Python in `PATH`
+        run: tox -e unit
+
+  artifact-update:
+    name: Google Artifact Registry container update
+    runs-on: ubuntu-latest
+    needs: [tox-python-build]
+    env:
+      DOCKER_REGISTRY_NAME: us-central1-docker.pkg.dev
+      PYTORCH_IMAGE_NAME: us-central1-docker.pkg.dev/${{ secrets.GCP_PROJECT_ID }}/cas-services-cicd/cas-pytorch
+      PYTORCH_CUDA_IMAGE_NAME: us-central1-docker.pkg.dev/${{ secrets.GCP_PROJECT_ID }}/cas-services-cicd/cas-pytorch-cuda
+    # Notice it's running only when a tagged commit is pushed
+    if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags')
```

Comment:
Why not just make this a separate workflow triggered by the event we want?  And then leaving unit_test.yml as is (was?)

Reply:
It was my initial plan, but then I realized that we would have separate runs in the GitHub action tab. Also, I couldn't find a way how to it depended on unit tests. In the current implementation, docker images wouldn't be built if the tests failed.